pwndbg/gdblib/disasm/riscv.py

```diff
@@ -28,15 +29,13 @@ def _is_condition_taken(
         else:
             src2_unsigned = 0
 
-        if self.architecture == "rv32":
-            src1_signed = src1_unsigned - ((src1_unsigned & 0x80000000) << 1)
-            src2_signed = src2_unsigned - ((src2_unsigned & 0x80000000) << 1)
-        elif self.architecture == "rv64":
-            src1_signed = src1_unsigned - ((src1_unsigned & 0x80000000_00000000) << 1)
-            src2_signed = src2_unsigned - ((src2_unsigned & 0x80000000_00000000) << 1)
-        else:
+        if self.architecture not in {"rv32", "rv64"}:
+            # TODO: figure out what this check is for. Are there other riscv arch types in GDB?
```

Comment:
cc: @patryk4815 may know.

Reply:
if gdb don't detect, he will put just `riscv` type without rv32/rv64.
```
set architecture riscv
riscv       riscv:rv32  riscv:rv64 
```

Eg at `gdb remote` start you need to force `set arch riscv:rv32` or `set arch riscv:rv64` in some cases.

Reply:
I see - good to know!

Reply:
Perhaps we should add some comment near the `self.architecture` definition to inform developers on that thing?

Reply:
or: tell the user here in the error we raise they have to set arch riscv:rv32 or rv64?

Reply:
Did some more digging: it looks like our detection for the architecture will fallback to `rv64` on the side of Pwndbg if it cannot determine 32 v 64 bit.

https://github.com/pwndbg/pwndbg/blob/b74fcdde97a4cd4c6c072235040eb21ee7a42a5b/pwndbg/gdblib/arch.py#L97-L101

Some kind of alert to the user that the architecture could not be completely determine would be good.

Reply:
I added a comment in our arch detection function regarding this.